### DISCUSSION TITLE
Contract termination

### DISCUSSION THREAD

**Discussion by ununhexium** [eclipse-dataspace-protocol-base/DataspaceProtocol#17] [Ideas]
# Contract termination

This is a proposal to add support for contracts termination in the IDS protocol.

## Motivation

One of our clients requested a contract to be terminated before its planned date as set by its policy.

Contracts termination is a mechanism that is available in most contracts, where either party may terminate it under various conditions.

sovity implemented this contract termination here, but it's working as a workaround, and we believe that this feature would be best if available as a standard IDS state.

https://github.com/sovity/edc-ce/tree/main/extensions/contract-termination

But this should be a feature that is supported at the protocol level, as it's a feature common to all contracts, be it physical or virtual.

In real life, these contract termination may be subject to terms and conditions which could be implemented as policies but this is an area that we didn't explore.

## Implementation idea

This should not be implemented as a new state in the contract's negotiation state machine.

This could be implemented as an event system for contract agreements, as discussed in this Tractus-X thread:

https://github.com/eclipse-tractusx/tractusx-edc/discussions/1392#discussioncomment-10014438

--- 

Would it be possible to add such a feature in the IDS specification such that what is a quasi-ubiquitous IRL contract feature be available for everyone?

**Comment by arnoweiss** [comment]
Where should this eventing system be specified then? I suppose that would be a new spec project on top of DSP which then requires extension via profiles with dataspace-specific semantics.

**Reply by arnoweiss** [reply]
I've also heard ideas that this could be implemented via a new protocol describing the state machine and lifecycle of the [odrl:Agreement](https://eclipse-dataspace-protocol-base.github.io/DataspaceProtocol/#contract-agreement-message) that results from the [Contract Negotiation Protocol](https://eclipse-dataspace-protocol-base.github.io/DataspaceProtocol/#contract-negotiation-protocol).
